### PR TITLE
server: exit tidb-server directly when all connections not in txn

### LINKED ISSUE #44951: Exit tidb-server directly when all connections not in txn
## Enhancement
After https://github.com/pingcap/tidb/pull/32111,  user need to wait for all connections to be disconnected, or execute a new sql before tidb-server exit.

For example
1. start a tidb-server
2. use mysql client connect it
3. ctrl+c tidb-server. It will not exit direclty until timeout

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #44951

Problem Summary: After https://github.com/pingcap/tidb/pull/32111, user need to wait for all connections to be disconnected, or execute a new sql before tidb-server exit. It's not an expected behaviour.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Exit tidb-server directly when all connections not in txn
```

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -327,9 +326,6 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 		}
 	}
 
-	// Init rand seed for randomBuf()
-	rand.Seed(time.Now().UTC().UnixNano())
-
```
Comment (xhebox) [inline server/server.go:332]: What?

File: server/server.go [hunk 2]
```diff
@@ -917,6 +911,9 @@ func (s *Server) DrainClients(drainWait time.Duration, cancelWait time.Duration)
 	go func() {
 		defer close(allDone)
 		for _, conn := range conns {
+			if !conn.getCtx().GetSessionVars().InTxn() {
```
Comment (xhebox) [inline server/server.go:914]: I am afrait that this is a compatibility breaker. /cc @djshow832

File: server/server.go (hunk 1 above)
Comment (Defined2014) [reply server/server.go:332]: https://pkg.go.dev/math/rand#Seed

> If Seed is not called, the generator is seeded randomly at program startup.

Not sure do we need it anymore.

File: server/server.go (hunk 2 above)
Comment (Defined2014) [reply server/server.go:914]: The behaviour is same as https://github.com/pingcap/tidb/pull/32111 merged before, like v6.5.x

**Review by tiancaiamao — APPROVED** [review]

File: server/server.go (hunk 2 above)
Comment (djshow832) [reply server/server.go:914]: Why do you think it's a compatibility breaker? @xhebox

Comment (xhebox) [reply server/server.go:914]: It is different from 7.1, but same as 6.5.

**Review by xhebox — APPROVED** [review]

**Pull request merged**